src/Microsoft.FeatureManagement/FeatureFilters/TimeWindowFilter.cs

```diff
@@ -56,7 +56,19 @@ public Task<bool> EvaluateAsync(FeatureFilterEvaluationContext context)
                 return Task.FromResult(false);
             }
 
-            return Task.FromResult((!settings.Start.HasValue || now >= settings.Start.Value) && (!settings.End.HasValue || now < settings.End.Value));
+            //
+            // Hit the first occurrence of the time window
+            if ((!settings.Start.HasValue || now >= settings.Start.Value) && (!settings.End.HasValue || now < settings.End.Value))
+            {
+                return Task.FromResult(true);
+            }
+
+            if (settings.Recurrence != null)
+            {
+                return Task.FromResult(RecurrenceEvaluator.MatchRecurrence(now, settings));
```

Comment:
I don't think the error messages mention the feature name anywhere at the moment in `RecurrenceEvaluator`, so maybe we could pass that in here from `context.FeatureName` and add it to the error message to make understanding the exceptions easier.

Reply:
Sounds good. Maybe we should also change the behavior of the targeting filter. The targeting evaluator will not provide the feature name info in the error message either.

Reply:
Oh I see, yea not sure if there was a reason for not including it but I'd think the same would apply there.